frontend/src/app/download/download.component.ts

```diff
@@ -40,66 +55,61 @@ export class DownloadComponent implements OnChanges {
 
     constructor(
         private downloadService: DownloadService,
-        private notificationService: NotificationService
-    ) {}
-
-    get downloadDisabled(): boolean {
-        return !this.resultOverview || this.resultOverview.resultsCount === 0;
+        private notificationService: NotificationService,
+        private searchService: SearchService,
+        private authService: AuthService,
+        private router: Router,
+    ) {
+        this.userDownloadLimit = this.authService.getCurrentUser()?.downloadLimit;
+        this.downloadLimit = this.userDownloadLimit || this.directDownloadLimit;
     }
 
-    ngOnChanges() {
-        this.availableCsvFields = _.filter(this.corpus?.fields, 'downloadable');
-        const highlight = this.resultOverview?.highlight;
-        // 'Query in context' becomes an extra option if any field in the corpus has been marked as highlightable
-        if (highlight !== undefined) {
-            this.availableCsvFields.push({
-                name: 'context',
-                description: `Query surrounded by ${highlight} characters`,
-                displayName: 'Query in context',
-                displayType: 'text_content',
-                csvCore: false,
-                hidden: false,
-                sortable: false,
-                searchable: false,
-                downloadable: true,
-                filterOptions: null,
-                mappingType: null,
-            } as unknown as CorpusField);
+    ngOnChanges(changes: SimpleChanges) {
+        if (changes.corpus) {
+            this.availableCsvFields = _.filter(this.corpus?.fields, 'downloadable');
+            this.selectedCsvFields = _.filter(this.corpus?.fields, 'csvCore');
+        }
+        if (changes.queryModel) {
+            this.totalResults?.complete();
+            this.resultsConfig?.complete();
```

Comment:
Careful closing of the observables here, I wouldn't have thought to do so if replacing them with a new stream.
Out of curiosity, did problems arise when not closed?

Reply:
No, this is more "good practice", because we're creating observables in `ngOnChanges`, it's good to close them when new input comes in.

But this is triggered when the input receives a new `queryModel`. That's really an edge case. (See https://github.com/UUDigitalHumanitieslab/I-analyzer/issues/1247#issuecomment-1932377736 for more detail.)